Competition om modal adjustments/yuusuf

Ganja0003 opened the pull request:
What I changed

- Changed the position of the content sections to match the Figma design.

- Fetched and showed the number of created entries for the competition.

- Updated styles to be closer to the Figma design.

- Fixed some small style issues.

- Added scroll if the content is too long and overflows the modal.

siderdk commented on app/components/modals/CompetitionDetailsModal.jsx:
```diff
@@ -20,6 +21,15 @@ export default function CompetitionDetailsModal({ competitionId }) {
         })
         .finally(() => setLoading(false));
+
+      fetch(`/api/entries?competitionId=${competitionId}`)
```
to keep it consistent, please use the axios API

siderdk reviewed (changes requested):
Really cool job Yuusuf. very well done. 
If you could just use axios API instead of fetch, and then find a way to freeze/disable the background while the modal is showing. (in a previous project, I just displayed a layer that covered the page, but feel free to use other strategies). 
Once that done you can merge I think.

Ganja0003 commented:
Thanks for the feedback Soheib.
I have added scroll prevention for arrowUp, arrowDown, pageUp, pageDown, and space when the modal is open.
I also switched from fetch to axios for consistency, as you mentioned.

siderdk reviewed (approved):
Solid job buddy, very well done
